std/internal/math/gammafunction.d

```diff
@@ -111,7 +111,14 @@ real gammaStirling(real x)
     }
     else {
         w = 1.0L + w * poly( w, SmallStirlingCoeffs);
-        y = pow( x, x - 0.5L ) / y;
+        if ( (floatTraits!(real).realFormat == RealFormat.ieeeDouble) && (x > 143.0L) ) {
```

Comment:
You think it is better to do a `static if` on the first condition?
I think I chose against it, because of having to duplicate the `y = pow( x, x - 0.5L ) / y;`, but I have no preference. The alternative could be:
```
    else {
        w = 1.0L + w * poly( w, SmallStirlingCoeffs);
        static if (floatTraits!(real).realFormat == RealFormat.ieeeDouble) {
            // Avoid overflow in pow() for 64-bit reals
            if (x > 143.0L) ) {
                real v = pow( x, 0.5L * x - 0.25L );
                y = v * (v / y);
            }
            else {
                y = pow( x, x - 0.5L ) / y;
            }
        }
        else {
            y = pow( x, x - 0.5L ) / y;
        }
    }
```